src/hooks/useStreaming.ts

```
@@ -0,0 +1,315 @@
+          id: generateMessageId(),
+          type: 'assistant',
+          content: preservedContent,
+          timestamp: new Date(),
+          streaming: true,
+        };
+        
+        // Add the message to the conversation - check for existing streaming messages first
+        onUpdateMessages(prev => {
+          // Remove any existing streaming messages to prevent duplicates
+          const filteredMessages = prev.filter(msg => !msg.streaming);
+          return [...filteredMessages, assistantMessage!];
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)